### PR TITLE
Replaced single socket.io use with fetch

### PR DESCRIPTION
**Pull request by jonahdlin**
Also added infrastructure for eventually doing live updates. More has to be fleshed out for this to work.

### CODE REVIEW

File: src/client/services/cdc/index.js [hunk 1]
```diff
@@ -2,26 +2,14 @@ const io = require('socket.io-client');
...
-        updateFunction(cyJSON);
-      } catch(err) {
-        if (msg.length > 0) {
-          console.log(JSON.parse(msg));
-          //console.log(msg);
-        } else {
-          throw err;
-        }
+  getGraphAndLayout(uri, version) {
+    return fetch(`http://${window.location.hostname}/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, {
```
Comment (d2fong) [inline src/client/services/cdc/index.js]: Does the following code not work?
```js
fetch(`/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, ... );
```

Comment (d2fong) [reply src/client/services/cdc/index.js]: You should not be using window at all when possible unless you like programming with global variables.

File: src/client/services/cdc/index.js [hunk 2]
```diff
@@ -2,26 +2,14 @@ const io = require('socket.io-client');
...
+        'Accept': 'application/json'
       }
-    });
-  },
-
-  // Request a graph from the server. Should send back a cyJSON graph
-  requestGraph(uri, version) {
-    socket.emit('getLayout', {uri: uri, version: version.toString()});
+    }).then(res => res.json())
+    .then(data => JSON.parse(data));
```
Comment (d2fong) [inline src/client/services/cdc/index.js]: Redundant code.  res.json() already parses the response body.

https://developer.mozilla.org/en-US/docs/Web/API/Body/json

File: src/client/services/cdc/index.js (hunk 1 above)
Comment (jonahdlin) [reply src/client/services/cdc/index.js]: I'll change it to that. It works. Thanks for the tip!

File: src/client/services/cdc/index.js (hunk 2 above)
Comment (jonahdlin) [reply src/client/services/cdc/index.js]: Resolved. Turns out we had left in a stringify on the server-side, which is why we still needed that on the frontend.

**Pull request merged**